### PR TITLE
Adding register

### PR DESCRIPTION
**Pull request by KuWojtow**
duplicate of #37 to change branch repository

### CODE REVIEW

**Comment by ddnowicki** [comment]
duplicate of #37 to change branch repository

File: src/core/authenticator.cpp [hunk 1]
```diff
@@ -8,4 +8,10 @@ Authenticator::Authenticator()
 bool Authenticator::authenticateUser(const std::string& username,
                                      const std::string& password) const {
   return database.loginCheck(username, password);
+}
+
+bool Authenticator::registerUser(const std::string& username,
+                                 const std::string& password,
+                                 const std::string& email) const {
```
Comment (patkaczm) [inline src/core/authenticator.cpp]: This one shouldn't be `const` since we are performing an action that significantly impacts code.

**Review by patkaczm — CHANGES_REQUESTED** [review]

File: src/core/core.cpp [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #include <vector>
 using json = nlohmann::json;
 
-std::vector<Product> getProducts(const json& jsonData) {
+const std::vector<Product> getProducts(const json& jsonData) {
```
Comment (patkaczm) [inline src/core/core.cpp]: make `const` methods, do not return `const` data.

**Review by patkaczm — CHANGES_REQUESTED** [review]

**Review by ddnowicki — APPROVED** [review]

File: src/core/core.cpp [hunk 3]
```diff
@@ -5,7 +5,7 @@
 #include <vector>
 using json = nlohmann::json;
 
-std::vector<Product> getProducts(const json& jsonData) {
+std::vector<Product> const getProducts(const json& jsonData) {
```
Comment (patkaczm) [inline src/core/core.cpp]: this also is `const` data return, 
`const` method declaration looks like this:
return_type method_name (arguments) `const`;

**Review by patkaczm — CHANGES_REQUESTED** [review]

**Review by patkaczm — APPROVED** [review]

**Pull request merged**